### DISCUSSION TITLE
Denial of WSS connection without auth header.

### DISCUSSION THREAD

**Discussion by Sachin4403** [xmidt-org/xmidt#74] [General]
Hi Team

I am aware Themis is a purely optional component in Xmidt cloud, but when we configured the Themis and due to any reason if Themis is not able to issue JWT to the CPE then CPE will be connecting to Talaria without auth header, and WSS connection will be established.

Now as a part of this discussion, I want to make Talaria purely configurable i.e. if Themis is configured in Talaria then Talaria will deny WSS connections that are coming without auth header so that this kind of CPEs won't be connected to Xmidt cloud. and we should create metrics out of it and log such CPEs in the logs for further actions.

cc: @schmidtw

**Comment by schmidtw** [comment]
@johnabass are you aware of how to configure Talaria to do this?

**Reply by Sachin4403** [reply]
@schmidtw This might be discussed in  https://discussion.xmidt.io/t/questions-around-themis-talaira-jwt/52/2, Unfortunately, I cannot access this link.

@joe94  Can you help me with the resolution

**Comment by mhekt** [comment]
@Sachin4403

I'm facing the same problem. Have you managed to find a solution?

**Reply by Sachin4403** [reply]
Hello @mhekt 

This now purely configurable in talaria, you can configure as per the need and the same is being mentioned in https://github.com/xmidt-org/talaria/pull/348

**Reply by mhekt** [reply]
Thank you, @Sachin4403. I found that my issue was due to using an outdated Talaria Docker image that didn't support the configuration.
